pkg/config/config.go

```diff
@@ -79,6 +80,13 @@ func getDefaultConfig(path string) types.BacalhauConfig {
 	defaultConfig.Update.CheckStatePath = filepath.Join(path, UpdateCheckStatePath)
 	defaultConfig.Auth.TokensPath = filepath.Join(path, TokensPath)
 
+	// We default to the folder which contains the job store, and add
+	// a subfolder for the network store.
+	defaultConfig.Node.Network.StoreDir = filepath.Join(
+		filepath.Dir(defaultConfig.Node.Requester.JobStore.Path),
+		NetworkTransportStore,
+	)
+
```

Comment:
this is acceptable as temporary solution, but I am hoping we get rid of the newly added `Network.StoreDir` config and cli flags. I am happy to tackle this in a following PR